test/framework/machine_helpers.go

```diff
@@ -127,7 +127,13 @@ func GetControlPlaneMachinesByCluster(ctx context.Context, input GetControlPlane
 	Expect(input.ClusterName).ToNot(BeEmpty(), "Invalid argument. input.ClusterName can't be empty when calling GetControlPlaneMachinesByCluster")
 	Expect(input.Namespace).ToNot(BeEmpty(), "Invalid argument. input.Namespace can't be empty when calling GetControlPlaneMachinesByCluster")
 
-	options := append(byClusterOptions(input.ClusterName, input.Namespace), controlPlaneMachineOptions()...)
```

Comment:
AFAICT this never worked with multiple clusters as one label selector overrode the other - this ended up getting all machineDeployments in the namespace for me. Replaced it with a simpler inline definition of the label selector.

Reply:
Fun! nice catch!